pkg/route/api/validation/validation.go

```diff
@@ -179,3 +240,71 @@ func validateInsecureEdgeTerminationPolicy(tls *routeapi.TLSConfig, fldPath *fie
 
 	return nil
 }
+
+// validatePrivateKeyForCert checks if a private key PEM is valid and
+// is valid for the certificate (if any).
+func validatePrivateKeyForCert(keyPEM, certPEM string) error {
+	if err := validatePrivateKeyPEM(keyPEM); err != nil {
+		return err
+	}
+
+	if len(certPEM) > 0 {
+		if _, err := tls.X509KeyPair([]byte(certPEM), []byte(keyPEM)); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// validatePrivateKeyPEM checks if a private key PEM is valid.
+func validatePrivateKeyPEM(keyPEM string) error {
+	if _, err := validatePEMData(keyPEM); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// validateCertificatePEM checks if a certificate PEM is valid and
+// optionally verifies the certificate using the options.
+func validateCertificatePEM(certPEM string, options *x509.VerifyOptions) (*x509.Certificate, error) {
+	data, err := validatePEMData(certPEM)
+	if err != nil {
+		return nil, err
+	}
+
+	if data == nil || len(data.Bytes) < 1 {
+		return nil, fmt.Errorf("invalid/empty certificate data")
+	}
+
+	cert, err := x509.ParseCertificate(data.Bytes)
+	if err != nil {
+		return nil, fmt.Errorf("error parsing certificate: %s", err.Error())
+	}
+
+	if options != nil {
+		_, err = cert.Verify(*options)
+		if err != nil {
+			return cert, fmt.Errorf("error verifying certificate: %s", err.Error())
+		}
+	}
+
+	return cert, nil
+}
+
+// validatePEMData checks if the PEM data is valid.
+func validatePEMData(pemData string) (*pem.Block, error) {
+	block, rest := pem.Decode([]byte(pemData))
```

Comment:
there can be multiple PEM blocks in the cert, you can't assume the first one is the right one, or the only one. could this be a cert chain (multiple cert blocks)